model_compression_toolkit/common/mixed_precision/bit_width_setter.py

```diff
@@ -60,12 +60,20 @@ def set_bit_widths(quant_config: QuantizationConfig,
                                    fw_info,
                                    node,
                                    node_index_in_graph)
-            else:
-                # TODO: refactor after adding activation mixed precision
-                if node.is_activation_quantization_enabled():
-                    assert node.is_all_activation_candidates_equal, "Activation Mixed Precision is not supported"
-                    node.final_activation_quantization_cfg = node.candidates_quantization_cfg[
-                        0].activation_quantization_cfg
+            elif node.is_activation_quantization_enabled():
+                # If we are here, this means that we are in weights-only mixed-precision
+                # (i.e., activations are quantized with fixed bitwidth or not quantized)
+                # and that this node doesn't have weights to quantize
+                assert len(node.candidates_quantization_cfg) > 0, \
+                    "Node need to have at least one quantization configuration in order to quantize its activation"
+                node.final_activation_quantization_cfg = node.candidates_quantization_cfg[0].activation_quantization_cfg
+            elif node.is_weights_quantization_enabled():
```

Comment:
I had in mind that in a case where we have an activation-only node (say, relu) and we are in weight-only mp mode, then the node would have multiple candidates, all with the same activation's bitwidth.
But thinking about it again, I think that this case is not possible, since we check if weights quantization is enabled in order to set multiple candidates, so as you said, there will only be one candidate (I'll fix the assert)